Makefile.best_binaries

```diff
@@ -0,0 +1,45 @@
+#**************************************************************************
+#*                                                                        *
+#*                                 OCaml                                  *
+#*                                                                        *
+#*            Gabriel Scherer, projet Parsifal, INRIA Saclay              *
+#*                                                                        *
+#*   Copyright 2019 Institut National de Recherche en Informatique et     *
+#*     en Automatique.                                                    *
+#*                                                                        *
+#*   All rights reserved.  This file is distributed under the terms of    *
+#*   the GNU Lesser General Public License version 2.1, with the          *
+#*   special exception on linking described in the file LICENSE.          *
+#*                                                                        *
+#**************************************************************************
+
+# This Makefile should be included.
+
+# It expects:
+# - Makefile.common to be included as well
+# - a ROOTDIR variable pointing to the repository root
+#   relative to the including Makefile
+
+# It exports definitions of BEST_OCAML{C,OPT,LEX,DEP} commands that
+# run to either the bytecode binary built in the repository or the
+# native binary, if available. Note that they never use the boot/
+# versions: we assume that ocamlc, ocamlopt, etc. have been run first.
+
+check_not_stale = \
+  $(if $(shell test $(ROOTDIR)/$1 -nt $(ROOTDIR)/$2 && echo stale), \
+    $(info Warning: we are not using the native binary $2 \
+because it is older than the bytecode binary $1; \
+you should silence this warning by either removing $2 \
+or rebuilding it (or `touch`-ing it) if you want it used.), \
+    ok)
+
+choose_best = \
+  $(if $(and $(wildcard $(ROOTDIR)/$2), $(strip $(call check_not_stale,$1,$2))), \
```

Comment:
The formatting is introducing spaces at the start of `BEST_CAMLC` which is what's breaking the Windows build. You either have to put this on one big line, and really annoy @shindere, or do this:

```make
  $(if $(and $(wildcard $(ROOTDIR)/$2), $(strip \
  $(call check_not_stale,$1,$2))),$(ROOTDIR)/$2,$(CAMLRUN) $(ROOTDIR)/$1)
```
which keeps the lines under 80 chars, but breaks in a less readable place, or do this piece of GNU make hackery...

```make
  $(if $(and $(wildcard $(ROOTDIR)/$2), $(strip $(call check_not_stale,$1,$2))),$\
    $(ROOTDIR)/$2,$\
    $(CAMLRUN) $(ROOTDIR)/$1)
```

(or you have to put some Windows special case stuff in to convert the forward slashes to back slashes)

Sorry about that 🙁

Reply:
What about just adding a call to `strip`? Would the following work?

```
choose_best = $(strip $(if \
   $(and $(wildcard $(ROOTDIR)/$2),$(strip $(call check_not_stale,$1,$2))), \
    $(ROOTDIR)/$2, \
    $(ROOTDIR)/$1)))
```

Reply:
(I just pushed the `strip` version for CI golf.)

Reply:
Hah! Too long a day, clearly 🙂